hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenBurnHandler.java

```diff
@@ -84,22 +81,7 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
         final var op = txn.tokenBurnOrThrow();
-        final var fungibleCount = op.amount();
-        final var serialNums = op.serialNumbers();
-
-        validateTruePreCheck(op.hasToken(), INVALID_TOKEN_ID);
-
-        // If a positive fungible amount is present, the NFT serial numbers must be empty
-        validateFalsePreCheck(fungibleCount > 0 && !serialNums.isEmpty(), INVALID_TRANSACTION_BODY);
-
-        validateFalsePreCheck(fungibleCount < 0, INVALID_TOKEN_BURN_AMOUNT);
-
-        // Validate the NFT serial numbers
-        if (fungibleCount < 1 && !serialNums.isEmpty()) {
-            for (final var serialNumber : op.serialNumbers()) {
-                validateTruePreCheck(serialNumber > 0, INVALID_NFT_ID);
-            }
-        }
+        burnPureChecks(op.amount(), op.serialNumbers(), op.hasToken(), INVALID_TOKEN_BURN_AMOUNT);
```

Comment:
same here. Probably need to rename method to say its common for both burn and wipe

Reply:
Done